### DISCUSSION TITLE
Accessing data from cache in Next.js

### DISCUSSION THREAD

**Discussion by kacluk123** [vercel/swr#651] [General]
I have a problem with accessing data from cache. I have my dahboard component. Data fetching here is ok, and i can access it.

```
const Dashboard = ({ code }) => {
  const { 
    data, 
    error, 
    mutate 
  } = useSWR(['/api/user', code], (url, code) => api.auth.getAuthData({ code }))
  return (
    <MainLayout>
      <div>
        dasdasdasasd
      </div>
    </MainLayout>
  )
}
```
The problem starts when i'm trying to access this data in other component.
```
const Navbar = () => {
  const { 
    data, 
    error, 
    mutate 
  } = useSWR(['/api/user'])
  console.log(data)
  return (
    <Styled.Navbar>
      dasdasdas
    </Styled.Navbar>
  )
}
```
Data in navbar component is undefinied. Any reason why this is not working? Whole app is wrapped into SWR provider.

**Comment by sergiodxa** [comment]
You need to use the same key, in your second case you are missing the code as the second element of the array

**Reply by kacluk123** [reply]
But in navbar i don't have this code.  I was thinking that this is stored in cache by `key: /api/user`. Is there a way to access this data without providing parameter again?

**Reply by sergiodxa** [reply]
There is no, each key is unique so the string only is a different key, even the string is differ than an array with the string, you need to match the same exact key, this is because you could use the same first element of the array with a different second element and you will want it to be a different key in the cache.

You will need to somehow make the code available for the navbar.
